snuba/web/rpc/storage_routing/routing_strategies/storage_routing.py

```diff
@@ -320,7 +408,31 @@ def get_routing_decision(self, routing_context: RoutingContext) -> RoutingDecisi
         with sentry_sdk.start_span(op="decide_tier") as span:
             try:
                 routing_context.timer.mark(_START_ESTIMATION_MARK)
-                routing_decision = self._get_routing_decision(routing_context)
+
+                routing_context.allocation_policies_recommendations = (
+                    self._get_recommendations_from_allocation_policies(routing_context)
+                )
+                combined_allocation_policies_recommendations = (
+                    self._get_combined_allocation_policies_recommendations(routing_context)
+                )
```

Comment:
this is a weird flow. You're setting a field on the routing context just to pass that into another function to collapse them into a single decision. This makes it so that ` _get_combined_allocation_policies_recommendations` only works in sequence with `_get_recommendations_from_allocation_policies` because they use the RoutingContext to share state. The function signature of `_get_combined_allocation_policies_recommendations` is really:


```python
def _get_combined_allocation_policies_recommendations(self, policy_recommendations: List[QuotaAllowance]):
    pass
```

what you're doing removes boundaries on the scope of that function's responsibility